### DISCUSSION TITLE
Deb packages are not listed with release assets

### DISCUSSION THREAD

**Discussion by franciozzy** [Netatalk/netatalk#1823] [Ideas]
**Describe the bug**
The [Installing Netatalk on Debian Linux](https://github.com/Netatalk/netatalk/wiki/Installing-Netatalk-on-Debian-Linux) wiki page says that:

> The recent release series at the time of writing, we distribute Debian deb packages built for Bookworm amd64 together with Debian releases. See the assets attached with releases on GitHub.

However, no `.deb` assets can be found under https://github.com/Netatalk/netatalk/releases.

Is the wiki out of date or am I missing something obvious? 😕

**Comment by rdmark** [comment]
Ah yes, the wiki page was outdated. I have updated it now.

The last release that I published a deb package for was [4.0.0](https://github.com/Netatalk/netatalk/releases/tag/netatalk-4-0-0).

After that I pivoted to working on the `netatalk` deb to be reintroduced into Debian repos proper, which has [since been successful](https://tracker.debian.org/pkg/netatalk). When Debian Trixie releases mid 2025, it will be distributed with netatalk again.

And the reason this is relevant, is because as part of the packaging process, I split up the `netatalk` package into 11 atomic packages, to better represent the sprawling suite of networking tools that is netatalk. Manually testing and distributing 11 debs on GitHub is too much of a manual overhead for our release process, so I stopped...

If you have a suggestion for safe and low-overhead distribution mechanisms for debs, I'd be curious to learn more. :)

**Comment by franciozzy** [comment]
Thank you for fixing the wiki.

I'm afraid I'm embarrassingly hopeless at `.deb`; despite being a Debian user for a very long time, I have mostly worked with `.rpm`. Was the work done for 4.0.0 manual? The only suggestion I can think of is creating a `debian/` directory in the project and including all the bits for package generation. But I'm guessing that's non-trivial?

**Comment by rdmark** [comment]
Oh, so if you look closely at the wiki page now, I've added a link to the Debian Salsa repository where the deb packaging sources are stored:

https://salsa.debian.org/netatalk-team/netatalk

If you clone this repo, you'll get the netatalk code plus the `debian/` dir. Building a deb is as easy as:

1. Set up a Debian dev environment: https://www.debian.org/doc/manuals/debmake-doc/ch04.en.html
1. Install all the netatalk dependencies (see `debian/control`)
1. Download the latest netatalk release tarball and put it into the parent dir of the cloned netatalk repo
1. Run `debuild` (or Debian build wrapper of your choice)

If all goes well, you'll get 11 fresh deb packages (for your current architecture) in the parent dir. Easy! :)
